src/controllers/namespace.go

```diff
@@ -19,12 +19,12 @@ type NamespaceController interface {
 }
 
 type namespaceController struct {
-	client *kubernetes.Clientset
+	client kubernetes.Interface
```

Comment:
Why is this change needed?

Reply:
It's needed in order to use `fake.Clientset` which implements `clientset.Interface` and not `kubernetes.Clientset`. See here:
```
k8s.io/client-go/kubernetes/fake/clientset_generated.go

// Clientset implements clientset.Interface. Meant to be embedded into a
// struct to get a default implementation. This makes faking out just the method
// you want to test easier.
type Clientset struct {
	testing.Fake
	discovery *fakediscovery.FakeDiscovery
	tracker   testing.ObjectTracker
}
```